### PR TITLE
Mpuncel/watch uuid pods

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

**Comment by mpuncel** [comment]
rebased on master

File: pkg/kp/kv.go [hunk 1]
```diff
@@ -388,50 +392,47 @@ func HealthPath(service string, node types.NodeName) string {
...
+			return ManifestResult{}, util.Errorf("Could not parse '%s' as pod index", pair.Key)
 		}
-	}
 
-	manifest, err := manifest.FromBytes(pair.Value)
-	if err != nil {
-		return ManifestResult{}, err
-	}
+		// TODO: add caching to pod store, since we're going to be doing a
+		// query per index now. Or wait til consul 0.7 and use batch fetch
```
Comment (rudle) [inline pkg/kp/kv.go]: consul 0.7 is gonna be so hype

**Comment by rudle** [comment]
:+1

**Comment by mpuncel** [comment]
rebased this on master and https://github.com/square/p2/pull/613, so review that one first

**Comment by mpuncel** [comment]
made a subtle but important change: `orchestrate.go` now filters out pods that have a non-nil PodUniqueKey because they are not expected to have a /reality entry. The implementation for these uuid pods will come in a later PR

**Comment by petertseng** [comment]
OK, I say 👍 . I did end up looking at everything.

**Pull request merged**